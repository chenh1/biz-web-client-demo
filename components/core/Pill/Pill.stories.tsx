import type { PillType } from './Pill';
import { Pill } from './Pill';

type Args = {
  args: PillType
}

export default {
  component: Pill
}

// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const HyperRedPill: Args = {
  args: {
    children: 'Some tag content',
    color: 'hyper-red'
  },
};

export const UltraPurplePill: Args = {
  args: {
    children: 'Some tag content',
    color: 'ultra-purple'
  },
};

export const CyberTealPill: Args = {
  args: {
    children: 'Some tag content',
    color: 'cyber-teal'
  },
};

export const ForceYellowPill: Args = {
  args: {
    children: 'Some tag content',
    color: 'force-yellow'
  },
};

export const MachIndigoPill: Args = {
  args: {
    children: 'Some tag content',
    color: 'mach-indigo'
  },
};

export const ApexBluePill: Args = {
  args: {
    children: 'Some tag content',
    color: 'apex-blue'
  },
};

