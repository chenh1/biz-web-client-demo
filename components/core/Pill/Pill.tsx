import type { CoreColors } from '../../constants/colors';
import type { GtmSet } from '../../constants/tagging'
import React, { useMemo } from 'react';
import { CLICK_CLASS, SECTION_VIEW_CLASS } from '../../constants/tagging';

const getColorClasses = (color: CoreColors): string => {
  switch(color) {
    case 'hyper-red': {
      return 'bg-hyper-red text-night-black'
    }
    case 'ultra-purple': {
      return 'bg-ultra-purple text-max-white'
    }
    case 'cyber-teal': {
      return 'bg-cyber-teal text-night-black'
    }
    case 'force-yellow': {
      return 'bg-force-yellow text-night-black'
    }
    case 'mach-indigo': {
      return 'bg-mach-indigo text-max-white'
    }
    case 'apex-blue': {
      return 'bg-apex-blue text-night-black'
    }
    default: {
      return "";
    }
  }
}

export type PillType = {
  children?: string | JSX.Element | JSX.Element[], 
  onClick?: (any) => void,
  color?: CoreColors, 
  className?: string, 
} & GtmSet

export const Pill: React.FC<PillType> = ({ children, onClick, color = "hyper-red", className="", gtmClick, gtmSectionView, dataGtmId }) => {
  const computedColorClasses = useMemo(() => {
    return getColorClasses(color);
  }, [color])

  return (
    <span data-gtm-id={dataGtmId} onClick={onClick} className={`${gtmClick ? CLICK_CLASS : ''} ${gtmSectionView ? SECTION_VIEW_CLASS : ''} cursor-pointer inline-block font-headings text-sm italic px-2 rounded-lg ${computedColorClasses} ${className}`}>
      {children}
    </span>
  )
};