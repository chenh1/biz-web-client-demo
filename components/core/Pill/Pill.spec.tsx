import { Pill } from "./Pill";
import { render } from '@testing-library/react';
import { JSDOM } from "jsdom"
const dom = new JSDOM()
global.document = dom.window.document
global.window = dom.window


it('renders a pill element', async () => {
  const { asFragment } = render(
    <Pill>Category A</Pill>
  );

  expect(asFragment()).toMatchSnapshot();
});
