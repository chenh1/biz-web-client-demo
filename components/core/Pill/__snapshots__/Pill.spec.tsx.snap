// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a pill element 1`] = `
<DocumentFragment>
  <span
    class="  cursor-pointer inline-block font-headings text-sm italic px-2 rounded-lg bg-hyper-red text-night-black "
  >
    Category A
  </span>
</DocumentFragment>
`;
